### PR TITLE
MOTRECALLS-1015 and 1016 - Check recall status for a vehicle receiving an MOT Reminder using the VRM

### PR DESCRIPTION
**Pull request by alexisc-kainos**
## Description 1015

Call the Recalls internal API using a vehicle’s VRM, in order to determine its recall status.

## Description 1016

If a subscriber to MOT Reminders is due to receive their first MOT notification (email or SMS) and they have an outstanding recall on their vehicle, a second notification for the recall should be sent alongside the MOT Reminder.

The MOT Reminders are first sent out:

For MOT vehicles (e.g cars):

1 month before expiry

For non MOT vehicles (e.g HGV/PSV):

2 months before expiry

The recall notification should only accompany the first MOT reminder and should not be sent alongside subsequent reminders (2 weeks before MOT and 1 day after MOT).

## Related issues

Related issue: https://dvsa.atlassian.net/browse/MOTRECALLS-1016
Related issue: https://dvsa.atlassian.net/browse/MOTRECALLS-1015

## Before submitting (or marking as "ready for review")

- [ ] Does the pull request title follow the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification?
- [ ] Have you performed a self-review of the code
- [ ] Have you have added tests that prove the fix or feature is effective and working
- [ ] Did you make sure to update any documentation relating to this change?

### CODE REVIEW

File: motr-data-mock/src/main/java/uk/gov/dvsa/motr/datamock/model/ByTestNumberResolver.java [hunk 1]
```diff
@@ -32,6 +36,33 @@ public Response resolve(String testNumber) {
                     .manufactureYear("2006")
                     .dueDate("2016-11-26")
                     .testNumber(testNumber);
+        } else if (testNumber.contains("999999")) {
```
Comment (iichr) [inline motr-data-mock/src/main/java/uk/gov/dvsa/motr/datamock/model/ByTestNumberResolver.java:39]: so they need added here too in addition to the byVrmResolver?

File: motr-data-mock/src/main/java/uk/gov/dvsa/motr/datamock/model/ByVrmResolver.java [hunk 2]
```diff
@@ -248,6 +248,28 @@ public Response resolve(String vrm) {
...
+                    .model("3008")
+                    .primaryColour("Silver")
+                    .vehicleType(MOT).manufactureYear("2006")
+                    .dueDate(daysFromToday(60))
+                    .testNumber("999999");
+        } else if (vrm.contains("AA33BBB")) {
+            response.make("VAUXHALL")
+                    .model("ASTRA")
+                    .primaryColour("Silver")
+                    .vehicleType(MOT).manufactureYear("2008")
```
Comment (iichr) [inline motr-data-mock/src/main/java/uk/gov/dvsa/motr/datamock/model/ByVrmResolver.java:263]: Would be good to have an example for a non-MOT vehicle too. Then again wouldn't all of those be redundant when/if the faker is gone

File: motr-webapp/webapp/src/main/java/uk/gov/dvsa/motr/web/system/binder/ConfigBinder.java [hunk 3]
```diff
@@ -18,9 +14,9 @@ public class ConfigBinder extends AbstractBinder {
...
     protected void configure() {
 
-        bindFactory(MotrConfigFactory.class).to(Config.class).in(Singleton.class);
-        bindFactory(AwsKmsDecryptorFactory.class).to(Decryptor.class);
-        bindFactory(VehicleDetailsClientFactory.class).to(VehicleDetailsClient.class);
-        bindFactory(CookieCipherFactory.class).to(CookieCipher.class);
+        bindFactory(uk.gov.dvsa.motr.web.system.binder.factory.MotrConfigFactory.class).to(Config.class).in(Singleton.class);
+        bindFactory(uk.gov.dvsa.motr.web.system.binder.factory.AwsKmsDecryptorFactory.class).to(Decryptor.class);
+        bindFactory(uk.gov.dvsa.motr.web.system.binder.factory.VehicleDetailsClientFactory.class).to(VehicleDetailsClient.class);
+        bindFactory(uk.gov.dvsa.motr.web.system.binder.factory.CookieCipherFactory.class).to(CookieCipher.class);
```
Comment (iichr) [inline motr-webapp/webapp/src/main/java/uk/gov/dvsa/motr/web/system/binder/ConfigBinder.java:20]: The fully qualified imports again :D

**Review by iichr — APPROVED** [review]

File: motr-data-mock/src/main/java/uk/gov/dvsa/motr/datamock/model/ByVrmResolver.java (hunk 2 above)
Comment (alexisc-kainos) [reply motr-data-mock/src/main/java/uk/gov/dvsa/motr/datamock/model/ByVrmResolver.java:263]: Sure, do we have an example non-mot vehicle in the faker? 
If/when the faker is gone, these will just act like the existing vehicles in this data mock, no recalls will be returned from them

File: motr-data-mock/src/main/java/uk/gov/dvsa/motr/datamock/model/ByTestNumberResolver.java (hunk 1 above)
Comment (alexisc-kainos) [reply motr-data-mock/src/main/java/uk/gov/dvsa/motr/datamock/model/ByTestNumberResolver.java:39]: They are aren't they. The test number mocked in ByVrmResolver is used in ByTestNumberResolver

File: motr-data-mock/src/main/java/uk/gov/dvsa/motr/datamock/model/ByVrmResolver.java (hunk 2 above)
Comment (iichr) [reply motr-data-mock/src/main/java/uk/gov/dvsa/motr/datamock/model/ByVrmResolver.java:263]: Done now :)

File: motr-data-mock/src/main/java/uk/gov/dvsa/motr/datamock/model/ByTestNumberResolver.java (hunk 1 above)
Comment (iichr) [reply motr-data-mock/src/main/java/uk/gov/dvsa/motr/datamock/model/ByTestNumberResolver.java:39]: makes sense, thanks :)

**Review by StuartKainos — APPROVED** [review]

**Comment by iichr** [comment]
QA completed and evidence attached to each of the tickets:
- https://dvsa.atlassian.net/browse/MOTRECALLS-1015?focusedCommentId=385145
- https://dvsa.atlassian.net/browse/MOTRECALLS-1016?focusedCommentId=385127

**Review by DariuszJerzewski — APPROVED** [review]

**Pull request merged**